### PR TITLE
[MB-1201] Modified method in getting user roles when adding a topic in event component through UI

### PR DESCRIPTION
**Pull request by hemikak**
JIRA - https://wso2.org/jira/browse/MB-1201

There was an issue of using a String array for getting the user roles and removing admin and anonymous users.

Therefore now using an Array list and iterator to remove those roles.

### CODE REVIEW

File: components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java [hunk 1]
```diff
@@ -444,30 +447,34 @@ private boolean validateTopicName(String topicName) {
...
-                          CarbonConstants.REGISTRY_ANONNYMOUS_ROLE_NAME.equals(role))) {
-                        rolesExceptAdminRole[index] = role;
-                        index++;
+                // check if more roles available than admin role and anonymous role
+                List<String> allRolesArrayList = new ArrayList<>();
+                Collections.addAll(allRolesArrayList, allRoles);
+
+                Iterator<String> it = allRolesArrayList.iterator();
+                while (it.hasNext()) {
+                    String nextRoles = it.next();
```
Comment (callkalpa) [inline components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java]: It seems it.next() returns a role, why the variable name a nextRoles?

File: components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java [hunk 2]
```diff
@@ -444,30 +447,34 @@ private boolean validateTopicName(String topicName) {
...
+                    if (nextRoles.equals(adminRole) || CarbonConstants.REGISTRY_ANONNYMOUS_ROLE_NAME.equals(nextRoles)) {
```
Comment (callkalpa) [inline components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java]: For consistency, shall we have nextRole.equals() in both cases?

File: components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java (hunk 1 above)
Comment (hemikak) [inline components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java]: Sorry that was a typo. Will fix it.

File: components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java (hunk 2 above)
Comment (hemikak) [inline components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java]: Sure, will make the necessary changes.

File: components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java [hunk 3]
```diff
@@ -444,30 +447,34 @@ private boolean validateTopicName(String topicName) {
     @Override
     public String[] getBackendRoles() throws EventBrokerException {
         UserRealm userRealm = CarbonContext.getThreadLocalCarbonContext().getUserRealm();
+        String[] cleanedRoles = new String[0];
```
Comment (callkalpa) [inline components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java]: In a case of there's only admin role, is this method supposed to return new String[0] ?

File: components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java [hunk 4]
```diff
@@ -37,10 +37,13 @@
 import org.wso2.carbon.user.api.UserRealm;
 import org.wso2.carbon.user.api.UserStoreException;
 import org.wso2.carbon.user.api.UserStoreManager;
+import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
```
Comment (callkalpa) [inline components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java]: Seems there's no usages

File: components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java (hunk 3 above)
Comment (hemikak) [inline components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java]: Yes, Admin role is not expected from this method. It has been described in the parent class comment.

**Pull request merged**